lib/cocoapods/target/build_settings.rb

```diff
@@ -602,15 +602,17 @@ def initialize(target, test_spec = nil)
         define_build_settings_method :libraries, :memoized => true, :sorted => true, :uniqued => true do
           return [] if (!target.requires_frameworks? || target.static_framework?) && !test_xcconfig?
 
-          libraries = libraries_to_import.dup
+          libraries = vendored_static_libraries.map { |l| File.basename(l, l.extname).sub(/\Alib/, '') }
```

Comment:
this functionality is duplicated, please extract it into its own `build_settings_method` so it only needs to be written & computed once?